packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/useGridRowGroupingPreProcessors.ts

```diff
@@ -17,6 +17,7 @@ import {
   createRowTree,
```

Comment:
@MBilalShafi 

> On the [codesandbox](https://codesandbox.io/s/hardcore-mahavira-n88l5j?file=/demo.tsx) where the package for this PR is installed, I played around and I observed that the page was not anymore stuck on row expansion change, but when I tried to change an aggregation function, it blocked the UI thread for a similar amount of time:

> I think the reason is that our [diffing algorithm](https://github.com/mui/mui-x/blob/e9cb67d6ce3c61d6e7b89ba44abd5e059c9e653f/packages/grid/x-data-grid-premium/src/hooks/features/aggregation/useGridAggregation.ts#L100-L123) is not smart enough to check, "ok, so out of 100 columns, the aggregation functions for column 1 has changed, let's go ahead and only re-compute the aggregation lookup for this column", rather it recomputes all the data once any column updates it's aggregation method.

That's a very good point, I'm sure we can optimize aggregation (especially when there's a lot of aggregated columns) 👍